stan/math/prim/mat/prob/bernoulli_logit_glm_lpmf.hpp

```diff
@@ -0,0 +1,181 @@
+
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
+#include <stan/math/prim/scal/meta/partials_return_type.hpp>
+#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/scal/err/check_bounded.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <stan/math/prim/scal/fun/constants.hpp>
+#include <stan/math/prim/scal/fun/log1m.hpp>
+#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <boost/random/bernoulli_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
+#include <cmath>
+#include <string>
+
+using Eigen::Dynamic;
+using Eigen::Matrix;
+
+namespace stan
+{
+namespace math
+{
+
+/**
+     * Returns the log PMF of the Generalized Linear Model (GLM)
+     * with Bernoulli distribution and logit link function.
+     * If containers are supplied, returns the log sum of the probabilities.
+     * @tparam T_n type of integer parameter
+     * @tparam T_x type of the data matrix (design matrix);
+     * this should be an Eigen::Matrix type
+     * @tparam T_beta type of the weight vector
+     * @tparam T_alpha type of the bias vector
+     * @param n integer parameter
+     * @param x design matrix
+     * @param beta weight vector
+     * @param alpha bias vector (in logits)
+     * @return log probability or log sum of probabilities
+     * @throw std::domain_error if theta is infinite.
+     * @throw std::invalid_argument if container sizes mismatch.
+     */
+
+template <bool propto,
+          typename T_n,
+          typename T_x,
+          typename T_beta,
+          typename T_alpha>
+typename return_type<T_x, T_beta, T_alpha>::type
+bernoulli_logit_glm_lpmf(const T_n &n,
+                         const T_x &x,
+                         const T_beta &beta,
+                         const T_alpha &alpha)
+{
+  static const std::string function = "bernoulli_logit_glm_lpmf";
+  typedef typename stan::partials_return_type<T_n,
+                                              T_x,
+                                              T_beta,
+                                              T_alpha>::type
+      T_partials_return;
+
+  using stan::is_constant_struct;
+  using std::exp;
+
+  if (!(stan::length(n) &&
+        stan::length(x) &&
+        stan::length(beta) &&
+        stan::length(alpha)))
+    return 0.0;
+
+  T_partials_return logp(0.0);
+
+  check_bounded(function, "n", n, 0, 1);
+  check_not_nan(function, "Data matrix", x);
+  check_not_nan(function, "Weight vector", beta);
+  check_not_nan(function, "Bias vector", alpha);
+  check_consistent_sizes(function,
+                         "Random variable", n,
+                         "Bias vector", alpha);
+  // check_consistent_sizes(function,
```

Comment:
Don't leave commented-out code in the code base.

Reply:
OK. I added all the relevant size checks now.